app/models/concerns/statistics/products_and_services.rb

```diff
@@ -0,0 +1,23 @@
+module Statistics::ProductsAndServices
+  extend ActiveSupport::Concern
+
+  included do
+    def reviews_count
+      self.reviews.count
+    end
+
+    def aggregate_score
+      score = 0
+      if self.reviews.count > 0
+        self.reviews.each { |r| score += r.score }
```

Comment:
Maybe can simplify using `sum`? e.g.
```
score = reviews.sum(:&score)
```

Reply:
Updated!

Reply:
In the future for scaling and optimisation, can consider to save `aggregate_score`, and `reviews_count` as fields in the model and update the values only when `after_save` of the respective model, as computing these values for each requests maybe quite computational intensive when the volume is high, but for POC this is good enough;) This also allow sorting by fields when querying.

Reply:
Understood, thanks kah kong!